### ISSUE TITLE
Pipeline in sklearn

### ISSUE THREAD

**Issue by PolarsDude** [abstractqqq/polars_ds_extension#293]
Hi!

Is it possible to use the polars_ds pipeline framework into sklearn pipeline? Do you have any examples on this?

**Comment by PolarsDude** [comment]
Thanks! The only reason I would use sklearn pipeline for is the fact that it can avoid data leakage between training and test data. Can polars_ds pipeline handle data leakage?

**Comment by abstractqqq** [comment]
Leakage is handled by splitting the df into train and test and only fit the pipeline on train. That should be how you avoid leaking. Is there anything else the sklearn pipeline does to prevent leaking?

**Comment by PolarsDude** [comment]
Thanks. One last question. Assume that the data has been split into test and training: X_train, y_train, X_test, and y_test. We also have our Polars pipeline named bp. How would one use the Polars pipeline in combination with ML training/prediction in sklearn? Would it look something like this?

*  for training : model.fit(bp.fit(X_train),y_train)

* for prediction on test: model.predict(bp.fit(X_test))

**Comment by PolarsDude** [comment]
Is there any support for including a ml model in the pds pipeline, for example a xgboost? In a sklearn pipeline you could write:
pipeline = Pipeline("pds_pipeline", pds_pipe(), "model", xgb.xgboost()). For training: pipeline.fit(x_train, y_train)

I am not sure how to write this into a pure pds pipeline (without including your first example).

**Comment by PolarsDude** [comment]
Alright. Thanks =)

**Comment by abstractqqq** [comment]
Just curious why do you want to do it with pure PDS? There must be a good reason. Is it because you want native Polars execution? I think sometimes I get strong opinions and forget what users really want

**Comment by PolarsDude** [comment]
Yes, it would be cool with a native polars execution. Also, I am used to implement sklearn pipelines containing data transformation and scoring. But I see your point regarding seperation of these two.
